server/monitor.go

```diff
@@ -2606,33 +2609,26 @@ func (s *Server) Jsz(opts *JSzOptions) (*JSInfo, error) {
 		opts.Accounts = true
 	}
 
-	// Check if we want a response from the leader only.
-	if opts.LeaderOnly {
-		js, cc := s.getJetStreamCluster()
-		if js == nil {
-			// Ignore
-			return nil, fmt.Errorf("%w: no cluster", errSkipZreq)
-		}
-		// So if we have JS but no clustering, we are the leader so allow.
-		if cc != nil {
-			js.mu.RLock()
-			isLeader := cc.isLeader()
-			js.mu.RUnlock()
-			if !isLeader {
-				return nil, fmt.Errorf("%w: not leader", errSkipZreq)
-			}
-		}
-	}
-
 	jsi := &JSInfo{
 		ID:  s.ID(),
 		Now: time.Now().UTC(),
 	}
-	if !s.JetStreamEnabled() {
+
+	js := s.getJetStream()
+	if js == nil || !js.isEnabled() {
 		jsi.Disabled = true
 		return jsi, nil
 	}
-	accounts := []*jsAccount{}
+
+	js.mu.RLock()
+	isLeader := js.cluster == nil || js.cluster.isLeader()
+	js.mu.RUnlock()
+
+	if opts.LeaderOnly && !isLeader {
+		return nil, fmt.Errorf("%w: not leader", errSkipZreq)
+	}
+
+	var accounts []*jsAccount
 
 	s.js.mu.RLock()
```

Comment:
we could surely do these things using `js` and not `s.js` thus avoiding an extra lock?

Reply:
That would be nice indeed. Starts here and apparently until line 2649 with `*s.js.usageStats()`.